src/net_processing.cpp

```diff
@@ -779,6 +779,19 @@ void PeerLogicValidation::InitializeNode(CNode *pnode) {
         PushNodeVersion(pnode, connman, GetTime());
 }
 
+void PeerLogicValidation::ReattemptInitialBroadcast(CScheduler& scheduler) const
+{
+    std::set<uint256> unbroadcast_txids = m_mempool.GetUnbroadcastTxs();
+
```

Comment:
For a follow-up, I think we should sanity check that the transaction still exists in the mempool before rebroadcasting. If not, delete the entry from the unbroadcast set.

If we don't do this, then a logic error where the entry isn't removed from the unbroadcast set when the tx is removed from the mempool turns into a memory leak and a network spam.

(Removal logic in removeUnchecked() is pretty straightforward and looks fine in this PR, but adding this extra check does no harm. It'd prevent a bug like https://github.com/bitcoin/bitcoin/pull/18038#discussion_r407665982 from causing eternal network spam)

Reply:
yeah totally agree. also was planning this for follow up. I've noted as a todo here: https://github.com/bitcoin/bitcoin/pull/18807#issue-410374742. Resolving this conversation and will track there.